File: xabber/src/main/java/com/xabber/android/data/extension/xtoken/SessionVO.java
package com.xabber.android.data.extension.xtoken;

public class SessionVO {

    private String client;
    private String device;
    private String uid;
    private String ip;
    private String lastAuth;

    public SessionVO(String client, String device, String uid, String ip, String lastAuth) {
        this.client = client;
        this.device = device;
        this.uid = uid;
        this.ip = ip;
        this.lastAuth = lastAuth;
    }

    public String getClient() {
        return client;
    }

    public String getDevice() {
        return device;
    }

    public String getUid() {
        return uid;
    }

    public String getIp() {
        return ip;
    }

    public String getLastAuth() {
        return lastAuth;
    }
}

File: xabber/src/main/java/com/xabber/android/data/extension/xtoken/SessionVO.kt
package com.xabber.android.data.extension.xtoken

import com.xabber.android.ui.text.getSmartTimeTextForRoster
import com.xabber.xmpp.xtoken.ResultSessionsIQ
import java.util.*

data class SessionVO(
    val client: String,
    val device: String,
    val uid: String,
    val ip: String,
    val lastAuth: String
)

/**
 * First element of pair is current session
 * Second element of pair is lost of other sessions
 */
fun ResultSessionsIQ.getMainAndOtherSessions(
    currentSessionUid: String
): Pair<SessionVO, List<SessionVO>> =
    this.sessions.sortedByDescending { it.lastAuth }.let { iqSessions ->
        Pair(
            iqSessions
                .first { it.uid == currentSessionUid }
                .let { SessionVO(it.client, it.device, it.uid, it.ip, "online") },
            iqSessions
                .filter { it.uid != currentSessionUid }
                .map {
                    SessionVO(
                        it.client,
                        it.device,
                        it.uid,
                        it.ip,
                        Date(it.lastAuth).getSmartTimeTextForRoster()
                    )
                }
        )
    }